### DISCUSSION TITLE
error on getCliColumns()

### DISCUSSION THREAD

**Discussion by Tellie** [pawelsalawa/sqlitestudio#4661] [General]
Hello
I tried to compile is with OS/2
I use qmake 5.12.2 and gcc 12.2.0 

When compiling the sqlitestudiocli i get
 RCC: Warning: No resources in '../sqlitestudiocli.qrc'.

weakld: error: Unresolved symbol (UNDEF) '__Z13getCliColumnsv'.
weakld: info: The symbol is referenced by:
    G:\sqlitestudio\output\build\sqlitestudiocli\clicommand.obj
    G:\sqlitestudio\output\build\sqlitestudiocli\clicommandsql.obj
    G:\sqlitestudio\output\build\sqlitestudiocli\clicommandhistory.obj
    G:\sqlitestudio\output\build\sqlitestudiocli\clicommanddesc.obj
Ignoring unresolved externals reported from weak prelinker.
make: *** [../../../output/SQLiteStudio/sqlitestudiocli.exe] Error 1

Any idea??

**Comment by pawelsalawa** [comment]
Firstly, Qt 5.12 is no longer actively supported in SQLiteStudio. I cannot guarantee it will compile with this Qt version.

Anyway, it's not the reason of your problem. It's just I have never thought of this application to be built on OS/2. I'm positively surprised someone tries.

So...

The problem here is file https://github.com/pawelsalawa/sqlitestudio/blob/master/SQLiteStudio3/sqlitestudiocli/cliutils.cpp, where there are marcro conditions:

```
#if defined(Q_OS_WIN32)
...
#elif defined(Q_OS_UNIX)
...
#endif
```

Inside of them there are `getCliColumns()` and `getCliRows()` defined. There is no block defining these 2 methods for OS/2 and frankly speaking I have no idea how to implement them for OS/2.

Another challenge will be the actual condition to be used, because I see no `Q_OS_OS2` at https://doc.qt.io/qt-5/qtglobal.html, but I guess there is another, common symbol defined under OS/2, so it should be possible to use it. Still, actual implementation of these 2 methods is yet to be figured out.

**Reply by pawelsalawa** [reply]
I forgot to mention - these methods should return integer that is number of rows/columns (or in other words lines & characters per line) available in the terminal window.

**Reply by Tellie** [reply]
Hihi

Thankz for the fast reply on this 

We use indeed  (Q-OS_OS2)

Normally we follow the windows way, i will play around and see if I can find out what will be best...

We have also QT v6.2.4,

i will try that to, but think it will fail into same error..

Where to find the CMD that will use the Qt6

**Reply by pawelsalawa** [reply]
Actually, Qt6 is even worse at the moment :) SQLiteStudio was not ported to Qt6 yet.
SQLiteStudio 3.4 is currently compiled (and verified) only against Qt 5.15.
SQLiteStudio 3.5 will use Qt6.

**Comment by Tellie** [comment]
Hi

Ok, i where mistaking we use QT v5.15.2

Op wo 18 jan. 2023 22:10 schreef Paweł Salawa ***@***.***>:

**Comment by Tellie** [comment]
Hi

Look @ https://github.com/bitwiseworks/qt5-os2/tree/5.15.2

Op wo 18 jan. 2023 22:14 schreef Elbert Pol ***@***.***>:

**Reply by pawelsalawa** [reply]
Yeah, this version is alright.

The only missing part are these 2 methods that I mentioned above. I have no idea how to implement them on OS/2.
